src/api/tokenList/TokenListApi.ts

```diff
@@ -108,6 +122,55 @@ export class TokenListApiImpl extends GenericSubscriptions<TokenDetails[]> imple
     currentUserList.push(token)
     localStorage.setItem(storageKey, JSON.stringify(currentUserList))
   }
+
+  private persistTokensForNetwork(networkId: number): void {
+    const storageKey = TokenListApiImpl.getLocalStorageKey(networkId)
+    localStorage.setItem(storageKey, JSON.stringify(this._tokensByNetwork[networkId]))
+  }
+
+  private async updateTokenIds(networkId: number): Promise<void> {
+    // Set as updated on start to prevent concurrent queries
+    this.updatedIdsForNetwork.add(networkId)
+
+    // Sequentially fetch updated ids
+    const updatedTokenList = []
+    let failedToUpdate = false
+    for (let i = 0; i < this._tokensByNetwork[networkId].length; i++) {
```

Comment:
ubermeganit: Can we do a foreach instead?

Reply:
Can't (find out how to) `await` inside a forEach/map/reduce etc.

But I'll update it like Dima asked for anyway, this won't be a problem anymore.

Reply:
Yep, it's not supported. Usually it's a map+promiseAll

Reply:
https://github.com/gnosis/dex-react/pull/824/commits/e6c39f6217e0f46f98b0749edc3d0a4bab12425c